website/project/views/register.py

```diff
@@ -47,6 +55,222 @@ def node_register_page(auth, node, **kwargs):
     ret.update(_view_project(node, auth, primary=True))
     return ret
 
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_be_public_registration
+def node_registration_retraction_get(auth, node, **kwargs):
+    """Prepares node object for registration retraction page.
+
+    :return: serialized Node to be retracted
+    :raises: 400: BAD_REQUEST if registration already pending retraction
+    """
+
+    # node = kwargs['node'] or kwargs['project']
+    if node.pending_retraction:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid Request',
+            'message_long': 'This registration is already pending a retraction.'
+        })
+
+    return serialize_node(node, auth, primary=True)
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_be_public_registration
+def node_registration_retraction_post(auth, node, **kwargs):
+    """Handles retraction of public registrations
+
+    :param auth: Authentication object for User
+    :return: Redirect URL for successful POST
+    """
+
+    data = request.get_json()
+    try:
+        node.retract_registration(auth.user, data['justification'])
```

Comment:
Even better: handle the case where `justification` is missing from the request body and return a proper error response.